### ISSUE TITLE
Perhaps add a shorthand for `ResidueRing(R, x)`?

### ISSUE THREAD

**Issue by fingolfin** [Nemocas/AbstractAlgebra.jl#998]
It would be kinda nice to have a shorthand for `ResidueRing(R, x)`. This is certainly not highly important, but in interactive use at least, I'd find it quite convenient if I could abbreviate to something like one of these (I am *not* saying all of these should be added, to the contrary; but perhaps one of them could be):
- `R % x` -- this is akin to the GAP syntax `Integers mod 6` and is based on the idea that `set mod N := { a mod N : a\in set }`
- `mod(R, x)` -- same idea; but since `%` maps to `rem`, this is perhaps more faithful mathematically
-  `R / x` or `R // x`: based on the mathematical notation `R / xR` (note that I also would like to allow `x*R` to form the principal ideal, but since AA has no ideals, I am not asking for it here; moreover, even then, `R / x*R` would be parsed as `(R/x)*R`; and writing `R/(x*R)` is somewhat awkward)

**Comment by fieker** [comment]
On Wed, Aug 18, 2021 at 01:20:48AM -0700, Max Horn wrote:
> It would be kinda nice to have a shorthand for `ResidueRing(R, x)`. This is certainly not highly important, but in interactive use at least, I'd find it quite convenient if I could abbreviate to something like one of these (I am *not* saying all of these should be added, to the contrary; but perhaps one of them could be):
> - `R % x` -- this is akin to the GAP syntax `Integers mod 6` and is based on the idea that `set mod N := { a mod N : a\in set }`
> - `mod(R, x)` -- same idea; but since `%` maps to `rem`, this is perhaps more faithful mathematically
> -  `R / x` or `R // x`: based on the mathematical notation `R / xR` (note that I also would like to allow `x*R` to form the principal ideal, but since AA has no ideals, I am not asking for it here; moreover, even then, `R / x*R` would be parsed as `(R/x)*R`; and writing `R/(x*R)` is somewhat awkward)

The desired abbreviation is quo(R, x) to get a quotient ring.
(To me)
I'd try to stay clear of / in any form as e.g. using ideals it is not
clear if the quotient strucure of the ideal quotient is intended.

I thought we supported quo in all variants, but aparently not.

There is also the question about maps... for Z/nZ this can/ is handled
by coerce in and out or "lift" out. There there is lift with 1 argument
(ring is implied) or 2 there the ring is passed as well.

quo "should" in all cases also return the map. In general I think I
cannot encode the data in lift - even with 2 arguments.

**Comment by fingolfin** [comment]
Right, I forgot about `quo`; that also makes sense to me, I'd be happy with that.
